### ISSUE TITLE
Problem during build with flutter_webrtc package that the connectycube sdk uses

### ISSUE THREAD

**Issue by gargkanishk** [ConnectyCube/connectycube-flutter-sdk-releases#87]
I am encountering an error while trying to build my Flutter project. The error message is as follows:

**What went wrong:**
`
Execution failed for task ':flutter_webrtc:compileDebugKotlin'.
> Error while evaluating property 'filteredArgumentsMap' of task ':flutter_webrtc:compileDebugKotlin'
   > Could not resolve all files for configuration ':flutter_webrtc:debugCompileClasspath'.
      > Failed to transform android-104.5112.05.jar (com.github.webrtc-sdk:android:104.5112.05) to match attributes {artifactType=android-classes-jar, org.gradle.category=library, org.gradle.libraryelements=jar, org.gradle.status=release, org.gradle.usage=java-api}.
         > Could not find android-104.5112.05.jar (com.github.webrtc-sdk:android:104.5112.05).
           Searched in the following locations:
               https://jitpack.io/com/github/webrtc-sdk/android/104.5112.05/android-104.5112.05.jar`

**Steps to Reproduce:**
1.  add the latest connectycube package: 2.4.0 to dependencies
2.  build the flutter project using flutter run

**Expected Behavior**
The project should build successfully without any errors.

**Actual Behavior**
The build process fails with the error message mentioned above.

**Additional info**
Added maven { url 'https://jitpack.io' } repository to project-level build.gradle file.
Cleaned the project by running flutter clean command.
Tried running after changing many kotlin version

**Comment by hwasiq** [comment]
Recently I am also facing this flutter_webrtc issue in my flutter project whenever I use connectycube.  
![image](https://user-images.githubusercontent.com/126855749/222659412-b3b8ea6a-59da-4b8a-9609-2cea9b10c3d5.png)

**Comment by TatankaConCube** [comment]
can you reproduce these issues in our samples? have you done the command `pod install` or `pod update` in your project's `ios` directory after adding the connectycube_sdk dependency?

**Comment by hwasiq** [comment]
> can you reproduce these issues in our samples? have you done the command `pod install` or `pod update` in your project's `ios` directory after adding the connectycube_sdk dependency?

Yes, I tried these steps several number of times.
Flutter Clean
Flutter Pub get
Pod Install or Pod update

**Comment by gargkanishk** [comment]
I was able to reproduce the same error with the sample app provided by connectycube

**Comment by muhammadfarhanshakil** [comment]
In project level build.gradle, add:
ext.kotlin_version = '1.6.10' & classpath 'com.android.tools.build:gradle:7.1.2'

In gradle/wrapper/gradle-wrapper.properties, add:
distributionUrl=https\://services.gradle.org/distributions/gradle-7.4-all.zip

Make sure openjdk 17.0.6 is installed in your system, it will surely workout

**Comment by TatankaConCube** [comment]
@hwasiq please provide your `build.gradle` files (project level and app level). Additionally, provide here the output of the command `flutter doctor`.

**Comment by muhammadfarhanshakil** [comment]
@hwasiq try to increase minimum deployment version in general settings of xcode

**Comment by gargkanishk** [comment]
@TatankaConCube waiting for some assistance as well?

**Comment by hwasiq** [comment]
> @hwasiq please provide your `build.gradle` files (project level and app level). Additionally, provide here the output of the command `flutter doctor`.

I face this problem in my flutter ios app when I run app on simulator.

**Comment by TatankaConCube** [comment]
> I face this problem in my flutter ios app when I run app on simulator.

@hwasiq sorry for the wrong mention))) have you tried the solution from [this](https://github.com/ConnectyCube/connectycube-flutter-sdk-releases/issues/87#issuecomment-1453631464) answer by changing it to the `11.0`?

**Comment by TatankaConCube** [comment]
@hwasiq for running on the iOS simulators you may need to add some configs to your `Podfile` like the next lines in the `post_install` function:
```
      # https://github.com/flutter-webrtc/flutter-webrtc/issues/713
      if target.name == "flutter_webrtc" || target.name == "WebRTC-SDK"
        config.build_settings['ONLY_ACTIVE_ARCH'] = 'YES'
      end
```

see the worked example in our P2P Calls sample - https://github.com/ConnectyCube/connectycube-flutter-samples/blob/master/p2p_call_sample/ios/Podfile#L46

**Comment by hwasiq** [comment]
This error occur in my Android project. 
Execution failed for task ':flutter_webrtc:compileReleaseKotlin'.

In project level build.gradle in add this
ext.kotlin_version = '1.6.10' & classpath 'com.android.tools.build:gradle:7.1.2'

And in gradle/wrapper/gradle-wrapper.properties, i add this.
distributionUrl=https://services.gradle.org/distributions/gradle-7.4-all.zip

Java Version:  17.0.6
connectycube_sdk: 2.4.0
Flutter Version:  3.7.6

How I can resolve this issue.
![image](https://user-images.githubusercontent.com/126855749/225608504-f33b6e42-8b3b-4847-8f3c-e10db82f069c.png)

build.grdle file
![image](https://user-images.githubusercontent.com/126855749/225609340-92856824-6421-4976-a971-81fdbb9680e6.png)

**Comment by TatankaConCube** [comment]
@hwasiq have you tried the [solution](https://github.com/ConnectyCube/connectycube-flutter-sdk-releases/issues/87#issuecomment-1453742228)? just for clarification - the issue is not at the Connectycube side, the issue related to the repo storage [problems](https://github.com/flutter-webrtc/flutter-webrtc/issues/1184#issuecomment-1450340277)

**Comment by TatankaConCube** [comment]
@gargkanishk @hwasiq right now we released the new version of our SDK ([2.4.1](https://pub.dev/packages/connectycube_sdk/versions/2.4.1)) it should resolve your issue with the Android build. Please try it and close this ticket if the problem is resolved.
